spec/api-net-spec.js

```diff
@@ -1619,6 +1619,29 @@ describe('net module', () => {
       `)
     })
   })
+
+  describe('Logging API', () => {
+    const fs = require('fs')
+    const os = require('os')
+    const path = require('path')
+    const dumpFile = path.join(os.tmpdir(), 'net_log.json')
+
+    afterEach(() => {
+      try {
+        fs.unlinkSync(dumpFile)
+      } catch (e) {
+        // Ignore error
+      }
+    })
+
+    it('should begin and stop logging to file when .startLogging() and .stopLogging() is called', (done) => {
+      net.startLogging(dumpFile)
+      net.stopLogging(() => {
```

Comment:
The [plan is to Promisify all of Electron's APIs](https://github.com/electron/electron/issues/9882), so I don't think it makes sense to introduce new APIs with callback interface. This should return a promise that can be `await`ed.

Reply:
LOL yea the current native implementation does a callback since it's more straightforward to implement on the C++ end. Once we find a good place for these API methods to live I think I can expose `.stopLogging()` as an async method.

One concern I have is not to result in a big mix of some methods utilizing callbacks and some returning promises.

cc: @MarshallOfSound

Reply:
Actually I just realized `.stopLogging([callback])` can return void when callback is provided, otherwise a promise. I'm not sure if that overloads too much complexity for the API.

Reply:
I've discussed this in many places, basically until we have a good consistent story for promises in cpp land I want to stick with callbacks.  My work on promise helpers in cpp has been stilted somewhat due to other more pressing concerns.